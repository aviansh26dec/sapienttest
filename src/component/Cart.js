import React from 'react';
import _ from 'lodash';

import cart from '../assets/images/cart.png';

import { Container, Navbar, Button, Badge, Image, OverlayTrigger, Popover, Table } from 'react-bootstrap';

function Cart({ update, products }) {
  let total = 0;
  let totalAmount = 0;
  let currency = '';

  if(products.length > 0){
    products.map((data) => {
      total += data.total

      if(parseInt(data.total) > 0){
        totalAmount += (parseInt(data.total) * parseInt(data.price));
      }
      currency = data.currency;
    })
  }

  const popover = (
    <Popover id="popover-basic">
      <Popover.Header as="h3">Cart</Popover.Header>
      <Popover.Body>
        <Table  bordered  className="mt-3">
            <thead>
                <tr>
                <th>#</th>
                <th>Title</th>
                <th>Action</th>
                </tr>
            </thead>
            <tbody>
                {products.length > 0 &&
                    products.map((item, index) => 
                        item.total > 0 ?
                          _.times(item.total, (keyValue) => {
                            return(<tr key={item.id}>
                                <td>{++keyValue}</td>
                                <td>{item.title}</td>
                                <td><Button size="sm" onClick={() => update('remove', index)}>Remove</Button></td>
                            </tr>)
                          })
                        :
                          index === 0 &&
                            <tr key="emptyCart">
                              <td colSpan={3}>Cart is empty.</td>
                            </tr>
                    )
                }
                <tr key="cartTotal">
                  <td colSpan={2}>Total Amount</td>
                  <td>{currency}{totalAmount}</td>
                </tr>
            </tbody>
        </Table>
      </Popover.Body>
      
    </Popover>
  );

  return (
    <>
      <Navbar className="bottomBorder">
        <Container>
          <Navbar.Brand href="#home">Prodct List</Navbar.Brand>
          <Navbar.Toggle />
          <Navbar.Collapse className="justify-content-end">
            <Navbar.Text>
              <OverlayTrigger trigger="click" placement="bottom" overlay={popover} rootClose={true}>
                <Button variant="success">
                  <Image src={cart} className="cartLogo" /> <Badge bg="danger">{total}</Badge>
                </Button>
              </OverlayTrigger>
            </Navbar.Text>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </>
  );
}

export default Cart;