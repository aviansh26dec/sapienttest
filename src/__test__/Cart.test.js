import { render, screen, wrapper } from '@testing-library/react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import Cart from '../component/Cart';

const product = {
    "products": [
      {
        "id": "123442",
        "title": "Product 1",
        "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor",
        "image": "/product1.jpeg",
        "price": "39",
        "currency": "$"
      },
      {
        "id": "123443",
        "title": "Product 2",
        "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor",
        "image": "/product2.jpeg",
        "price": "39",
        "currency": "$"
      }
    ]
  };

describe("Cart", () => {
  it('renders without crashing', () => {
      const div = document.createElement("div");
      ReactDom.render(<Cart products={product} />, div);
  });

  it('renders correctly', () => {
    const tree = renderer
    .create(<Cart products={product} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
