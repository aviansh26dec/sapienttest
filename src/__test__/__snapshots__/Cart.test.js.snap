// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders correctly 1`] = `
<nav
  className="bottomBorder navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Prodct List
    </a>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <span
        className="navbar-text"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <img
            className="cartLogo"
            src="cart.png"
          />
           
          <span
            className="badge bg-danger"
          >
            0
          </span>
        </button>
      </span>
    </div>
  </div>
</nav>
`;
